platform/broadcom/saibcm-modules/make/Make.config

```diff
@@ -173,6 +173,9 @@ CXXFLAGS += ${INCFLAGS}
 CPPFLAGS += ${INCFLAGS}
 
 CFLAGS += -DSAI_FIXUP -UKCOM_FILTER_MAX -DKCOM_FILTER_MAX=1024
+ifeq ($(ASIC_TYPE),DNX)
+	CFLAGS += -DBCM_DNX_SUPPORT -DBCM_DNX2_SUPPORT -DBCM_DNXF_SUPPORT
```

Comment:
checked the code, seems it does not matter. we should be able always build with the flag and install the driver for both xgs and dnx devices.

Reply:
Most places are ok. 

The only place which was pointed out by BRCM was here https://github.com/Azure/sonic-buildimage/blob/master/platform/broadcom/saibcm-modules/systems/linux/kernel/modules/bcm-knet/bcm-knet.c#L170, the usage of force_tagged flag. But I find this too is safe, as this check is under if (device_is_sand(sinfo)). 

Will get a confirmation from BRCM and update.

Reply:
thanks

Reply:
@lguohan, I had a discussion with Broadcom team on using a common driver for DNX and XGS. There suggestion was to keep it separate, they use it like that and don't have a history of using common modules. One of the main reason was that DNX and XGS for the same SAI release could  be using different **SDK VERSION**. This could create problems in future if it happens as usually GPL/kernel module release is tied to SDK version. ( Please find more details in CSP CS00012049832 )

I will try out something similar to what you suggested for sai bcm library by installing both the modules and point to the right one in the rc.local on bootup. This will save time than install when DUT is coming up.

Please share your comments.